inst/include/mode/stepper.hpp

```diff
@@ -120,6 +120,16 @@ class stepper {
     }
   }
 
+  void set_state(const stepper<Model>& other) {
+    k2 = other.k1;
+    k3 = other.y;
+  }
+
+  void swap() {
+    std::swap(k1, k2);
+    std::swap(y, k3);
```

Comment:
I see 3 different ways of assigning the contents of one vector to another now; this, the one on [line 142](https://github.com/mrc-ide/mode/pull/13/files#diff-36514440b7a7f854235117d7428c9ccb67791d9d6b14e8419a1c4dca0ac3a16eL132) and simple [reassignment](https://github.com/mrc-ide/mode/pull/13/files#diff-36514440b7a7f854235117d7428c9ccb67791d9d6b14e8419a1c4dca0ac3a16eR124) - which should be used when and why?

Reply:
so would it make sense to always use `copy_n` even if `n` is the whole vector length, rather than `=`?

Reply:
when I say "always", I actually mean quite specifically on lines 124/125, lol

Reply:
```
  void set_state(const stepper<Model>& other) {
    k2 = other.k1;
    k3 = other.y;
  }
```

here, yes - that would be better as copy_n for sure.  You could also use

```
  std::copy(other.k1.begin(), other.k1.end(), k2.begin());
  std::copy(other.y.begin(), other.y.end(), k3.begin());
```

see [`std::copy`](https://en.cppreference.com/w/cpp/algorithm/copy)